File: 221028-dojo/javascript/solution.js
export function countOne(number) {
  return number.toString(2).split('').filter((value) => value === '1').length;
}

export function countOneOfLargerNumber(standard, largerNumber) {
  const count = countOne(largerNumber);

  if (standard === count) {
    return largerNumber;
  }

  return countOneOfLargerNumber(standard, largerNumber + 1);
}

export default function solution(n) {
  const standard = countOne(n);

  return countOneOfLargerNumber(standard, n + 1);
}

File: 221028-dojo/java/app/src/main/java/Solution.java
class Solution {
  public int solution(int n) {
    int countNumberOfOne = countOne(n);

    int largerNumber = n + 1;

    while(true) {
      if (countNumberOfOne == countOne(largerNumber)) {
        return largerNumber;
      }
      largerNumber += 1;
    }
  }

  public int countOne(int number) {
    String binaryNumber = Integer.toBinaryString(number);

    int count = 0;

    for (int i = 0; i < binaryNumber.length(); i += 1) {
      if (binaryNumber.charAt(i) == '0') {
        continue;
      }
      count += 1;
    }

    return count;
  }
}
